src/compiler/crystal/macros/methods.cr

```diff
@@ -1533,6 +1538,10 @@ module Crystal
       ArrayLiteral.new(defs)
     end
 
+    def self.has_method?(type, name)
+      BoolLiteral.new(!!type.has_def?(name))
```

Comment:
right, sorry about that (found why: `some_hash.try &.has_key?(some_key)` can be `Nil` with the `try`..)